frontend/components/tree/js/tree.service.ts

```diff
@@ -1078,16 +1091,16 @@ export class TreeService {
 		while (nodes.length) {
 			const currentNode = nodes.pop();
 
-			const nodeIndex = this.currentSelectedNodes.indexOf(currentNode);
+			const nodeSelected = this.currentSelectedNodes[currentNode];
 
 			if (
 				select &&
 				currentNode.selected !== this.SELECTION_STATES.parentOfUnselected &&
-				nodeIndex === -1
+				!nodeSelected
```

Comment:
Seen this mutliple times now... try to avoid if statements where both conditions checks the same boolean.

````js
if(nodeSelected) {
    delete this.currentSelectedNodes[currentNode._id];
} else if (select && currentNode.selected !== this.SELECTION_STATES.parentOfUnselected) {
    this.currentSelectedNodes[currentNode._id] = currentNode;
}
````